agent/grpc-external/services/resource/watch.go

```diff
@@ -118,15 +115,33 @@ func (s *Server) validateWatchListRequest(req *pbresource.WatchListRequest) (*re
 		return nil, err
 	}
 
-	// Error when partition scoped and namespace not empty.
-	if reg.Scope == resource.ScopePartition && req.Tenancy.Namespace != "" {
-		return nil, status.Errorf(
-			codes.InvalidArgument,
-			"partition scoped type %s cannot have a namespace. got: %s",
-			resource.ToGVK(req.Type),
-			req.Tenancy.Namespace,
-		)
+	// Check scope
+	if err = validateScopedTenancy(reg.Scope, req.Type, req.Tenancy); err != nil {
+		return nil, err
 	}
 
 	return reg, nil
 }
+
+func getDefaultTenancy(scope resource.Scope) *pbresource.Tenancy {
```

Comment:
nit: since this is in the `resource` package, it will show up (e.g. auto complete) next all the other functions in the package and cause confusion.  Maybe `wildcardTenancyFor(...)` would read better.